gpflow/monitor/tensorboard.py

```diff
@@ -117,11 +117,9 @@ def _summarize_parameter(self, name: str, param: Union[Parameter, tf.Variable]):
                 "run the monitor outside the `tf.function`."
             )
 
-        if size == 1:
-            tf.summary.scalar(name, param[0], step=self.current_step)
-        else:
-            for i in range(min(size, self.max_size)):
-                tf.summary.scalar(f"{name}[{i}]", param[i], step=self.current_step)
+        it = range(size) if self.max_size == -1 else range(min(size, self.max_size))
+        for i in it:
+            tf.summary.scalar(f"{name}[{i}]", param[i], step=self.current_step)
```

Comment:
This changes behaviour (namely, it'll always add the `[0]` for scalar parameters). Might be nice to only have your change in the else: branch of the original code?

Reply:
Ah, makes sense! Sure I'll make the change